core/src/scene/scene.h

```diff
@@ -107,6 +108,8 @@ class Scene {
 
     std::shared_ptr<Texture> getTexture(const std::string& name) const;
 
+    std::atomic_ushort m_resourceLoad;
```

Comment:
Ok nevermind, atomic makes sense here (load and store are safe but increment can be a problem).